lib/concurrent-ruby/concurrent/atomic/mutex_semaphore.rb

```diff
@@ -23,8 +23,17 @@ def acquire(permits = 1)
 
       synchronize do
         try_acquire_timed(permits, nil)
-        nil
       end
+
+      if block_given?
+        begin
+          yield
+        ensure
+          release(permits)
+        end
+      end
+
+      nil
```

Comment:
I figured it was safe to not put this in the `synchronize` block, since we don't need the contents of the block to run in a synchronized context, and `release` is itself synchronized.

Reply:
It's there to stop an internal value being accidentally returned by the API - nothing to do with `synchronized`.